### ISSUE TITLE
APIv1 (clapi) poller reload isn't handled properly

### ISSUE THREAD

**Issue by d4nys3k** [centreon/centreon#1137]
# BUG REPORT INFORMATION

### Prerequisites

***Centreon 22.10.7***

### Description

When changing configuration and reloading poller using APIv1 (clapi), poller reports Conf Changed after issuing `applycfg`.

![image](https://user-images.githubusercontent.com/16295884/226864174-708b571a-5387-4c5c-a6dd-1f34daa50e84.png)

In clapi, there's probably missing call of `SENDCBCFG` command. 
Also, `updated` flag in `nagios_server` table isn't handled properly.

### Steps to Reproduce

`https://$CENTHOST/centreon/api/index.php?action=action&object=centreon_clapi" -H "Content-Type: application/json" -H "centreon-auth-token: $AUTHTOKEN" --data "{\"action\":\"applycfg\",\"values\":\"$POLLER\" }"`

### Describe the received result

Conf Changed = **yes**

### Describe the expected result

Conf Changed = **no**

### Logs

During onfiguration reload using clapi, `gorgoned.log` contains following command sequence:
```
2023-03-22 10:29:25 - INFO - [legacycmd] Handling command 'SENDCFGFILE', Target: '10'
2023-03-22 10:29:25 - INFO - [legacycmd] Handling command 'RELOAD', Target: '10'
2023-03-22 10:29:25 - INFO - [legacycmd] Handling command 'RELOADBROKER', Target: '10'
```

Command sequence in `gorgoned.log` using poller reload in GUI:
```
2023-03-22 10:37:22 - INFO - [legacycmd] Handling command 'SENDCFGFILE', Target: '10'
2023-03-22 10:37:22 - INFO - [legacycmd] Handling command 'SENDCBCFG', Target: '10'
2023-03-22 10:37:22 - INFO - [legacycmd] Handling command 'RELOAD', Target: '10'
```

### Additional relevant information (e.g. frequency, ...)

I thing adding `SENDCBCFG` command to `/usr/share/centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php` and modification of SQL query in `centreon.Config.Poller.class.php` should fix this issue (similar approach as found in `/usr/share/centreon/www/include/configuration/configGenerate/xml/moveFiles.php`, only if there's broker configuration present.
```
--- /usr/share/centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php.orig	2023-03-02 17:11:56.000000000 +0100
+++ /usr/share/centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php	2023-03-22 12:13:07.107668766 +0100
@@ -210,7 +210,7 @@
         $msg_restart = _("OK: A reload signal has been sent to '" . $host["name"] . "'");
         print $msg_restart . "\n";
         $statement = $this->DB->prepare(
-            "UPDATE `nagios_server` SET `last_restart` = :last_restart WHERE `id` = :poller_id LIMIT 1"
+            "UPDATE `nagios_server` SET `last_restart` = :last_restart, `updated` = '0' WHERE `id` = :poller_id LIMIT 1"
         );
         $statement->bindValue(':last_restart', time(), \PDO::PARAM_INT);
         $statement->bindValue(':poller_id', (int) $poller_id, \PDO::PARAM_INT);
@@ -634,6 +634,10 @@
                 );
             }
             exec("echo 'SENDCFGFILE:" . $host['id'] . "' >> " . $this->centcore_pipe, $stdout, $return);
+            $listBrokerFile = glob($this->brokerCachePath . '/' . $host['id'] . "/*.{xml,json,cfg}", GLOB_BRACE);
+            if (count($listBrokerFile) > 0) {
+                exec("echo 'SENDCBCFG:" . $host['id'] . "' >> " . $this->centcore_pipe, $stdout, $return);
+            }
 
             $msg_copy .= _(
                 "OK: All configuration will be send to '"
```

**Comment by BenoitPoulet** [comment]
The bug has been fixed, but not included in any release 🙃

https://github.com/centreon/centreon/pull/925/files

**Comment by d4nys3k** [comment]
> The bug has been fixed, but not included in any release upside_down_face
> 
> https://github.com/centreon/centreon/pull/925/files

What about external commands issued during reload?  `SENDCBCFG` isn't sent when reloading over clapi, but I see it in Gorgone log during reload issued over GUI.

**Comment by omercier** [comment]
Hi @d4nys3k, 
The fix will be released in the next releases (22.10.8 and 22.04.13, currently under validation).
Please feel free to reopen the issue if you still face the bug after the update.
